docs/paper/dev/getting-started/plugin-yml.md

```diff
@@ -0,0 +1,192 @@
+---
+slug: /dev/plugin-yml
+---
+
+# Paper Plugin YML
+
+The plugin.yml file is the main configuration file for your plugin. 
+It contains information about your plugin, such as its name, version, and description. 
+It also contains information about the plugin's dependencies, permissions, and commands.
+
+The plugin.yml file is located in the `resources` directory of your project.
+```
+example-plugin
+├── build.gradle.kts
+├── settings.gradle.kts
+└── src
+    └── main
+        ├── java
+        └── resources
+            └── plugin.yml
+```
+
+## Example
+
+Here is an example of a plugin.yml file:
+
+```yaml
+name: ExamplePlugin
+version: 1.0.0
+main: io.papermc.testplugin.ExamplePlugin
+description: An example plugin
+author: PaperMC
+website: https://papermc.io
+api-version: 1.19
+```
+
+## Fields
+
+:::note
+
+The fields in this section are not in any particular order. 
+If they have an asterisk (\*) next to them, that means they are required.
+
+:::
+
+### name*
+
+The name of your plugin. This is what will be displayed in the plugin list and log messages.
+- `name: ExamplePlugin`
+
+### version*
+
+The current version of the plugin. 
+- `version: 1.0.0`
+
+### main*
+
+The main class of your plugin. This is the class that extends `JavaPlugin` and is the entry point to your plugin.
+- `main: io.papermc.testplugin.ExamplePlugin`
+
+This is the package path and class name of your main class.
+
+### description
+
+A short description of your plugin and what it does.
+- `description: An example plugin`
+
+### author / authors
+
+The author(s) of the plugin. This can be a single author or a list of authors.
+- `author: PaperMC`
+- `authors: [PaperMC, SpigotMC, Bukkit]`
+
+### website
+
+The website of the plugin. This is useful for linking to a GitHub repository or a plugin page.
+- `website: https://papermc.io`
+
+### api-version
+
+The version of the Paper API that your plugin is using. This doesn't include the minor version.
+Servers with a version lower than the version specified here will refuse to load the plugin.
+The valid versions are 1.13 - 1.19.
+- `api-version: 1.19`
+
+:::info
+
+If this is not specified, the plugin will be loaded as a legacy plugin and a warning will be printed to the console.
+
+:::
+
+### load
+
+This tells the server when to load the plugin. This can be `STARTUP` or `POSTWORLD`.
+- `load: STARTUP`
+
+### prefix
+
+The prefix of the plugin. This is what will be displayed in the log instead of the plugin name.
+- `prefix: EpicPaperMCHypePlugin`
+
+### libraries
+
+:::warning
+
+There is debate over whether this is against maven central's TOS. This may be removed in the future.
+
+:::
```

Comment:
It is against their guidelines, but I don't think there is any plan from either paper or (and I obviously don't know definitively from this one) spigot. If anything, It'd just get moved over to a paper/spigot hosted caching proxy.